etcdserver/raft.go

```diff
@@ -52,6 +52,11 @@ const (
 )
 
 var (
+	// protect raftStatus
+	// raftStatus might be accessed concurrently 
+	// by integration tests since multiple etcdservers
+	// share the same raftStatus
+	raftStatusMu sync.Mutex
```

Comment:
This is here only for the integration tests. The raftstatus will only be accessed once. But for safety purpose, we can do more probably.